src/tools/async-download/components/AsyncDownloadForm.tsx

```diff
@@ -0,0 +1,208 @@
+  const [submitDisabled, setSubmitDisabled] = useState(() =>
+    // default ids value will tell us if submit should be disabled or not
+    isInvalid(
+      initialFormValues[AsyncDownloadFields.name].selected,
+      downloadUrlOptions
+    )
+  );
+  // used when the form is about to be submitted to the server
+  const [sending, setSending] = useState(false);
+  // flag to see if a title has been set (either user, or predefined)
+  const [jobNameEdited, setJobNameEdited] = useState(
+    // default to true if it's been set through the history state
+    Boolean(initialFormValues[AsyncDownloadFields.name].selected)
+  );
+
+  // extra job-related fields
+  const [jobName, setJobName] = useState(
+    initialFormValues[AsyncDownloadFields.name]
+  );
+
+  const submitAsyncDownloadJob = useCallback(
+    (event: FormEvent | MouseEvent) => {
+      event.preventDefault();
+
+      setSubmitDisabled(true);
+      setSending(true);
+
+      // navigate to the dashboard, not immediately, to give the impression that
+      // something is happening
+      sleep(1000).then(() => {
+        history.push(LocationToPath[Location.Dashboard]);
+
+        // We emit an action containing only the parameters and the type of job
+        // the reducer will be in charge of generating a proper job object for
+        // internal state. Dispatching after history.push so that pop-up messages (as a
+        // side-effect of createJob) cannot mount immediately before navigating away.
+        dispatchTools(
+          createJob(
+            { ...downloadUrlOptions, compressed: false, download: false },
+            JobTypes.ASYNC_DOWNLOAD,
+            jobName.selected
+          )
+        );
+      });
+    },
+    // NOTE: maybe no point using useCallback if all the values of the form
+    // cause this to be re-created. Maybe review submit callback in all 4 forms?
+    [history, dispatchTools, downloadUrlOptions, jobName]
+  );
+
+  useEffect(() => {
+    if (jobNameEdited) {
+      return;
+    }
+    const potentialJobName = getPotentialJobName(count, downloadUrlOptions);
+    setJobName((jobName: AsyncDownloadFormValue) => {
+      if (jobName.selected === potentialJobName) {
+        // avoid unecessary rerender by keeping the same object
+        return jobName;
+      }
+      return { ...jobName, selected: potentialJobName };
+    });
+  }, [count, downloadUrlOptions, downloadUrlOptions.namespace, jobNameEdited]);
+
+  useEffect(() => {
+    setSubmitDisabled(isInvalid(jobName.selected, downloadUrlOptions));
+  }, [downloadUrlOptions, jobName]);
+
+  return (
+    <form
+      onSubmit={submitAsyncDownloadJob}
+      aria-label="Async download job submission form"
+      ref={scrollRef}
+    >
+      <fieldset>
+        <section className="tools-form-section tools-form-section--compact">
+          <section className="tools-form-section__item tools-form-section__item--full-width">
+            <h4>Submit A Generate File Job</h4>
```

Comment:
```suggestion
            <h4>Submit A "Generate File" Job</h4>
```
I would suggest separating the type of job with the rest of the sentence, as someone might try to parse this sentence in plain English and it wouldn't make much sense